### DISCUSSION TITLE
How to map find results by default

### DISCUSSION THREAD

**Discussion by Coolroo** [SeaQL/sea-orm#1939] [Q&A]
Hi there, I am currently struggling to find how I can modify the results of a given column when using find. What I mean by this is that I want to be able to encrypt data going into the database, and then decrypt when pulling from the database. I can implement the former behaviour using ActiveModelBehaviour, but I am not sure how to get the latter behaviour. Does anyone have any docs they can point me to, or even know if this is possible?

I want to have this behaviour happen when creating a Model because I do not want to have to maintain this logic in multiple places in my codebase, because if I end up changing the behaviour I then have to track it down everywhere that I implemented it and I think it would be more elegant if it just came out in the state I described.

**Comment by tyt2y3** [comment]
One approach I have in mind:

Add an ignored column and a helper method e.g. 

```rust
Model {
    #[sea_orm(ignore)]
    decrypted: Option<T>,
}
```

and then the helper method has a signature of `fn get_decrypted(&mut self) -> &T` that decrypts the payload if not already decrypted
